internal/services/apimanagement/api_management_identity_provider_aad_resource.go

```diff
@@ -63,6 +63,13 @@ func resourceApiManagementIdentityProviderAAD() *pluginsdk.Resource {
 					ValidateFunc: validation.IsUUID,
 				},
 			},
+
+			"client_library": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 16),
```

Comment:
Where has this validation constraint come from?

Reply:
This is from [swagger](https://github.com/Azure/azure-rest-api-specs/blob/8322ff8ff0b9818c49e2e6ef7d02e3d0b0333839/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/definitions.json#L3149), details see below.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/d8200088-5497-4000-ae28-0bbb4d60ef32)